inst/tutorials/tidy-01-causality1/tidy-01-causality1.Rmd

```diff
@@ -0,0 +1,377 @@
+
+- `>` greater than, `>=` greater than or equal to
+- `<` less than, `<=` less than or equal to
+- `==` equal to
+- `!=` not equal to
+
+When we use these to compare two objects in R, we end us with a logical object. You can also compare a vector to a particular number.
+
+
+#### Exercises
+
+- Test if 10 is greater than 5.
+
+```{r gt-test, exercise=TRUE}
+
+```
+
+```{r gt-test-solution}
+10 > 5
+```
+
+```{r gt-test-check}
+grade_code()
+```
+
+- Test which values in the vector `x` is greater than or equal to 0.
+
+```{r compare, exercise = TRUE}
+## x vector
+x <- c(-2, -1, 0, 1, 2)
+
+## test which values of x are greater than or equal to 0
+
+```
+
+```{r compare-check}
+grade_result(
+  fail_if(~ identical(.result, x > 0),
+          "Did you forget the 'or equal to' part of the comparison?"),
+  pass_if(~ identical(.result, x >= 0))
+)
+```
+
+## Subsets in R
+
+### Subsetting a data frame
+
+
+You can use the same logical statements you have been using to create subsets of a data frame. These can often be helpful because we'll want to calculate various quantities of interest for different subsets of the data. For this exercise, we will use the `resume` data frame made up of the variables `firstname`, `sex`, `race`, and `call`. As a reminder, here is what the data look like:
+
+```{r}
+resume
+```
+
+#### Exercise
+
+- Use the `filter` function to create a subset of the `resume` data frame that is only female names that sound white. Save this subset as `resume.wf`
+- Use the `head` function to see the first six lines of this subset.
+- Use the `summarize` function to calculate the mean of the `call` variable in this subset. Save this output as `resume.wf.callback`.
+
+
+```{r subset, exercise = TRUE}
+## create the subset for white female names and
+## assign it to resume.wf
+resume.wf <- ...
+
+## print the first 6 lines of the subset
+
+
+## calculate the mean of the callback variable (call)
+
+```
+
+```{r subset-solution}
+resume.wf <- resume %>%
+  filter(race == "white" & sex == "female")
+
+head(resume.wf)
+
+resume.wf.callback <- resume.wf %>%
+  summarize(mean(call))
+
+resume.wf.callback
+```
+
+```{r subset-check}
+grade_code()
+```
+
+
+### Comparing means across treatment conditions
+
+
+You can use the same ideas as in the last step to create a different subset of the data corresponding to white-sounding female names. Then, you can compare the average callback for the white-female names to the average callback for the black-female names. This will give you a sense of how the employer callback rate varies by racial group of the applicant for females.
+
+#### Exercise
+
+- Create a subset of the `resume` data for black-sounding female names.
+- Calculate the mean of the `call` variable in this subset. Save this output as `resume.bf.callback`.
+- Print the difference in means between the `call` variable in the white-sounding name subset and the black-sounding name subset.
```

Comment:
Should we tell them to also do this for the `resume.wf` data frame? They may think it carries over from the last question. Or maybe it does?

Reply:
It does not carry over. I will rephrase the instructions!